cosmodc2/synthetic_subhalos/extend_subhalo_mpeak_range.py

```diff
@@ -185,19 +186,79 @@ def create_synthetic_lowmass_mock_with_satellites(
     return gals
 
 
-def get_redshifts_from_comoving_distances(comoving_distances, zmin, zmax, H0=71.0, OmegaM=0.2648):
+def get_comoving_distances(zmin, zmax, cosmology, H0=71.0):
+    rmin = (cosmology.comoving_distance(zmin)*H0/100.).value   #Mpc/h
+    rmax = (cosmology.comoving_distance(zmax)*H0/100.).value
+
+    return rmin, rmax
+
+
+def get_box_boundaries(Nside, cutout_id, rmin, rmax):
+
+    corners = hp.boundaries(Nside, cutout_id, nest=False)
+    center = np.asarray(hp.pixelfunc.pix2vec(Nside, cutout_id, nest=False)) # acccount for spherical cap
+    box_corners = np.vstack((corners.T*rmin, corners.T*rmax, center*rmax))
+    box_mins = np.asarray([np.min(box_corners[:, i]) for i in range(len(box_corners.T))])
+    box_maxs = np.asarray([np.max(box_corners[:, i]) for i in range(len(box_corners.T))])
+
+    return box_mins, box_maxs
+
+def get_volume_factor(box_mins, box_maxs, Nside, cutout_id, r_min, r_max, 
```

Comment:
There should really be a unit-test for this function.

Reply:
Yes, good catch,  I should have an additional test to check that volume_box is larger than volume_octant. If not, then the rest of the code in that function can be skipped and the volume_factor returned is 1.0